ext/standard/streamsfuncs.c

```diff
@@ -756,115 +782,132 @@ static int stream_array_emulate_read_fd_set(zval *stream_array)
 /* }}} */
 
 /* {{{ Runs the select() system call on the sets of streams with a timeout specified by tv_sec and tv_usec */
-PHP_FUNCTION(stream_select)
-{
-	zval *r_array, *w_array, *e_array;
-	struct timeval tv, *tv_p = NULL;
-	fd_set rfds, wfds, efds;
-	php_socket_t max_fd = 0;
-	int retval, sets = 0;
-	zend_long sec, usec = 0;
-	bool secnull;
-	bool usecnull = 1;
-	int set_count, max_set_count = 0;
-
-	ZEND_PARSE_PARAMETERS_START(4, 5)
-		Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
-		Z_PARAM_ARRAY_EX2(w_array, 1, 1, 0)
-		Z_PARAM_ARRAY_EX2(e_array, 1, 1, 0)
-		Z_PARAM_LONG_OR_NULL(sec, secnull)
-		Z_PARAM_OPTIONAL
-		Z_PARAM_LONG_OR_NULL(usec, usecnull)
-	ZEND_PARSE_PARAMETERS_END();
-
-	FD_ZERO(&rfds);
-	FD_ZERO(&wfds);
-	FD_ZERO(&efds);
-
-	if (r_array != NULL) {
-		set_count = stream_array_to_fd_set(r_array, &rfds, &max_fd);
-		if (set_count > max_set_count)
-			max_set_count = set_count;
-		sets += set_count;
-	}
-
-	if (w_array != NULL) {
-		set_count = stream_array_to_fd_set(w_array, &wfds, &max_fd);
-		if (set_count > max_set_count)
-			max_set_count = set_count;
-		sets += set_count;
-	}
+PHP_FUNCTION(stream_select) {
+    zval *r_array, *w_array, *e_array;
+    struct timeval tv, *tv_p = NULL;
+    fd_bigset rfds, wfds, efds;
+    php_socket_t max_fd = 0;
+    int retval, sets = 0;
+    zend_long sec, usec = 0;
+    bool secnull;
+    bool usecnull = 1;
+    int set_count, max_set_count = 0;
+
+    ZEND_PARSE_PARAMETERS_START(4, 5)
+        Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
+        Z_PARAM_ARRAY_EX2(w_array, 1, 1, 0)
+        Z_PARAM_ARRAY_EX2(e_array, 1, 1, 0)
+        Z_PARAM_LONG_OR_NULL(sec, secnull)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_LONG_OR_NULL(usec, usecnull)
+    ZEND_PARSE_PARAMETERS_END();
+
+    FD_BIGSET_ZERO(&rfds, 15000); // Adjust the number of file descriptors as needed
+    FD_BIGSET_ZERO(&wfds, 15000);
+    FD_BIGSET_ZERO(&efds, 15000);
+
+    if (r_array != NULL) {
+        set_count = stream_array_to_fd_bigset(r_array, &rfds, &max_fd);
+        if (set_count > max_set_count)
+            max_set_count = set_count;
+        sets += set_count;
+    }
 
-	if (e_array != NULL) {
-		set_count = stream_array_to_fd_set(e_array, &efds, &max_fd);
-		if (set_count > max_set_count)
-			max_set_count = set_count;
-		sets += set_count;
-	}
+    if (w_array != NULL) {
+        set_count = stream_array_to_fd_bigset(w_array, &wfds, &max_fd);
+        if (set_count > max_set_count)
+            max_set_count = set_count;
+        sets += set_count;
+    }
 
-	if (!sets) {
-		zend_value_error("No stream arrays were passed");
-		RETURN_THROWS();
-	}
+    if (e_array != NULL) {
+        set_count = stream_array_to_fd_bigset(e_array, &efds, &max_fd);
+        if (set_count > max_set_count)
+            max_set_count = set_count;
+        sets += set_count;
+    }
 
-	if (!PHP_SAFE_MAX_FD(max_fd, max_set_count)) {
-		RETURN_FALSE;
-	}
```

Comment:
We still need this if we initialize the fd sets to a fixed size

Reply:
The FD_BIGSET is now initialized dynamically, which was always my intention - just forgot that while I was eager to make the PR after seeing everything work.